### ISSUE TITLE
Question about merging hashtables

### ISSUE THREAD

**Issue by Outek** [AutomatedLab/DscWorkshop#13]
Hi

I have some troubles to merge hashtables from the baseline with hashtables from a role.
Let me explain my intention based on the DscWorkshop Files.

I would like to createa file on every Server. Now i have a specific role which i would like create another  file.

SOmething like here: https://github.com/AutomatedLab/DscWorkshop/blob/dev/DscSample/DSC_ConfigData/Roles/FileServer/FilesAndFolders/Items.yml#L2

In the Baseline: 
```powershell
- DestinationPath: C:\Test1.txt
  Type: File
  Contents: A file on every Server created with Dsc
```

In the Role:
```powershell
- DestinationPath: C:\Createthisfile.txt
  Type: File
  Contents: A file on a specific role created with Dsc
```
That's my Datum.yml configuration, but it don't merge the values.
The Default is:  default_lookup_options: MostSpecific

```powershell
FilesAndFolders: hash

FilesAndFolders\Items:
  merge_hash_array: DeepTuple
  merge_options:
    tuple_keys:
      - Destination
      - Type
      - Contents
```
Maybe you @gaelcolas can help me a little bit?
Thank you

**Comment by gaelcolas** [comment]
Hi,
I'm traveling and without a PC for me to try on atm, but here's how I'd troubleshoot this.

Load `$Node` with a relevant hashtable for that role in your session, load `$Datum` as well with the structure.
Do a `lookup 'FilesAndFolders' -Node $Node -Datum $Datum -Verbose` to see where and why the expected merge doesn't happen.

The output tells you the Datum Path and merging strategy it tries to merge, or whether the Path returned `$null`.

If you cannot solve this, please push your code to a new branch in your fork of this project, and link it here. I'll have a look next week.

**Comment by gaelcolas** [comment]
After re-reading, are you trying to merge items that are in separate files and treat them as array of hashtable? If so, it's a known limitation iirc, Datum doesn't type the object properly for merging. If you put in single yml file it works with your Datum definition, but with role as folders and items as files, the merge won't work.

**Comment by Outek** [comment]
This is what i'm trying to do..
https://github.com/AutomatedLab/DscWorkshop/pull/14

**Comment by raandree** [comment]
The PR gaelcolas/Datum#53 makes Datum support folders for roles and also should solve some merge issues I have came across. Can you give this branch a  try?

**Comment by gaelcolas** [comment]
This is fixed in Datum 0.0.34. Please raise an issue in the Datum project if any problem.
